lib/cloudant.js

```diff
@@ -75,7 +75,42 @@ Cloudant.prototype.getTypes = function() {
  */
 Cloudant.prototype.connect = function(cb) {
   debug('Cloudant.prototype.connect');
-  cb(null, this.cloudant);
+  var self = this;
+  // check if cloudant server is up
+  if (self.options.url) {
+    request.get(self.options.url, function(err, res, body) {
+      if (err) cb(res.errMsg);
+    });
+  } else {
+    var opts = {
+      auth: {
+        user: self.options.username,
+        pass: self.options.password,
+        sendImmediately: true,
+      },
+      uri: self.options.username + '.cloudant.com',
+      port: self.options.port || '',
+    };
+    request.get(opts, function(err, res, body) {
+      if (err) cb(res.errMsg);
+    });
+  }
+  // strip db name if defined in path of url before
+  // sending it to our driver
+  if (self.options.url) {
```

Comment:
Doing that made me realize that we won't get in the callback for `request.get` calls in time for using the cloudant connection object down the line, so I have removed it altogether.